### PR TITLE
SDIT-1108: ✨ Start adding micro frontend components

### PR DESCRIPTION
**Pull request by petergphillips**
- SDIT-1108: ✨ Start adding micro frontend components

### CODE REVIEW

File: docker-compose-test.yml [hunk 1]
```diff
@@ -16,8 +16,11 @@ services:
     - hmpps_int
     container_name: wiremock
     restart: always
+    command: "--global-response-templating --enable-stub-cors" # add command `--verbose` to output verbose request matching
```
Comment (petergphillips) [inline docker-compose-test.yml:19]: allows us to grab the request path and use it to find the files for the mock response

File: feature.env [hunk 2]
```diff
@@ -10,5 +10,5 @@ API_CLIENT_SECRET=clientsecret
 SYSTEM_CLIENT_ID=clientid
 SYSTEM_CLIENT_SECRET=clientsecret
 TOKEN_VERIFICATION_ENABLED=true
-FRONTEND_COMPONENTS_URL=http://localhost:9091
-FRONTEND_COMPONENTS_ENABLED=true
+FRONTEND_COMPONENTS_URL=http://localhost:9091/frontend-components
+FRONTEND_COMPONENTS_ENABLED=false
```
Comment (petergphillips) [inline feature.env:14]: temporarily disable frontend components as PR was getting too big

File: helm_deploy/values-dev.yaml [hunk 3]
```diff
@@ -22,7 +22,6 @@ generic-service:
     RESTRICTED_PATIENT_API_URL: https://restricted-patients-api-dev.hmpps.service.justice.gov.uk
     SUPPORT_URL: https://support-dev.hmpps.service.justice.gov.uk/
     FRONTEND_COMPONENTS_URL: https://frontend-components-dev.hmpps.service.justice.gov.uk
-    FRONTEND_COMPONENTS_ENABLED: true
```
Comment (petergphillips) [inline helm_deploy/values-dev.yaml:25]: temporarily disable frontend components in dev too

File: server/config.ts [hunk 4]
```diff
@@ -100,7 +100,7 @@ export default {
       enabled: get('TOKEN_VERIFICATION_ENABLED', 'false') === 'true',
     },
     frontendComponents: {
-      url: get('FRONTEND_COMPONENTS_URL', 'http://localhost:8083', requiredInProduction),
+      url: get('FRONTEND_COMPONENTS_URL', 'http://localhost:8085/frontend-components', requiredInProduction),
```
Comment (petergphillips) [inline server/config.ts:103]: default to our local docker wiremock instance

File: server/data/frontendComponentsClient.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  restClient: RestClient
+
+  constructor(token: string) {
+    this.restClient = new RestClient('Frontend Components', config.apis.frontendComponents, token)
+  }
+
+  async getComponents<T extends ComponentType[]>(userToken: string): Promise<Record<T[number], Component>> {
+    return this.restClient.get<Record<T[number], Component>>({
+      path: `/components`,
+      query: 'component=header&component=footer',
```
Comment (petergphillips) [inline server/data/frontendComponentsClient.ts:22]: grab both header and footer at same time

File: server/data/frontendComponentsClient.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      headers: { 'x-user-token': userToken },
```
Comment (petergphillips) [inline server/data/frontendComponentsClient.ts:23]: micro components uses the `x-user-token` rather than the authorisation header

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**